### PR TITLE
[ADO Reporter] Add ADO logging

### PR DESCRIPTION
**Pull request by Bryan-Cee**
This PR introduces ADO reporter from issue:
#167

### CODE REVIEW

File: lib/args.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+"use strict";
```
Comment (kenotron) [inline lib/args.js:1]: Please don't commit lib files in the repo

File: src/logger/initReporters.ts [hunk 2]
```diff
@@ -12,16 +13,23 @@ export function initReporters(config: Config) {
     logLevel = LogLevel[config.logLevel as LogLevelString];
   }
 
-  const reporters = [
+  const reporters: Array<AdoReporter | JsonReporter | NpmLogReporter> = [
```
Comment (kenotron) [inline src/logger/initReporters.ts:16]: Please add a "reporter" type to this via a string rather than using an environment variable.

**Review by kenotron — CHANGES_REQUESTED** [review]
Please clean this PR up a bit and re-request a PR

File: lib/args.js (hunk 1 above)
Comment (Bryan-Cee) [reply lib/args.js:1]: Sure thing, I was using this branch for testing that is why I committed the lib folder.

File: src/logger/initReporters.ts (hunk 2 above)
Comment (kenotron) [reply src/logger/initReporters.ts:16]: This is fine - in the future, we should think about how to clean this up. We should allow for multiple reporter types.

File: src/logger/reporters/AdoReporter.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import {getPackageAndTask} from '../../task/taskId';
+import {RunContext} from '../../types/RunContext';
+import {Reporter} from './Reporter';
+import {LogEntry} from '../LogEntry';
+
+export class AdoReporter implements Reporter {
+  log(entry : LogEntry) {}
```
Comment (kenotron) [inline src/logger/reporters/AdoReporter.ts:7]: Do you want to have more info per entry?

**Review by kenotron — APPROVED** [review]

**Pull request merged**